File: Ej 31 (Marco)/Java/Marcos.java
/**
 * @author DeathbatO
 */
public class Marcos {

    public static void mostrarMarcoRectangular(String texto) {
        // Dividir el texto en palabras
        String[] palabras = texto.split("\\s+");

        // Encontrar la longitud de la palabra más larga
        int longitudMaxima = 0;
        for (String palabra : palabras) {
            if (palabra.length() > longitudMaxima) {
                longitudMaxima = palabra.length();
            }
        }

        // Imprimir el marco superior
        System.out.println("*".repeat(longitudMaxima + 4));

        // Imprimir las palabras con el marco lateral
        for (String palabra : palabras) {
            System.out.printf("* %-" + longitudMaxima + "s *\n", palabra);
        }

        // Imprimir el marco inferior
        System.out.println("*".repeat(longitudMaxima + 4));
    }

    public static void main(String[] args) {
        String textoEjemplo = "Esta es una frase de ejemplo para probar la función.";
        mostrarMarcoRectangular(textoEjemplo);
    }
}

File: Ej 31 (Marco)/Python/Marcos.py
## DeathbatO

def mostrar_marco_rectangular(texto):
    # Dividir el texto en palabras
    palabras = texto.split()

    # Encontrar la longitud de la palabra más larga
    longitud_maxima = max(len(palabra) for palabra in palabras)

    # Imprimir el marco superior
    print('*' * (longitud_maxima + 4))

    # Imprimir las palabras con el marco lateral
    for palabra in palabras:
        print('* {:<{}} *'.format(palabra, longitud_maxima))

    # Imprimir el marco inferior
    print('*' * (longitud_maxima + 4))

# Ejemplo de uso
texto_ejemplo = "Esta es una frase de ejemplo para probar la función."
mostrar_marco_rectangular(texto_ejemplo)
